common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -85,7 +84,7 @@ protected Recycler() {
     }
 
     protected Recycler(int maxCapacity) {
-        this.maxCapacity = Math.max(0, maxCapacity);
+        this.maxCapacity = Math.min(0, maxCapacity);
```

Comment:
why this change ? This makes no sense as this will always result in 0 or a negative value